### PR TITLE
Adds charges to omens and omen smiting. Reduces omen bad luck if nobody's nearby.

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

refactor: Adds charges to omens and omen smiting rather than only being permanent or one-use. Mirrors now grant seven bad luckers.

qol: Reduces omen bad luck if nobody's nearby to witness the funny. (Ghosts are included in the check!)

fix: Fixed an issue where a monkey check in doorcrushing was never actually able to pass. Also they screech now.

## Why It's Good For The Game

> refactor: Adds charges to omens and omen smiting rather than only being permanent or one-use. Mirrors now grant seven bad luckers.

Allows for someone to get between 1-infinity omen accidents. Seriously why wasnt this a thing before

> qol: Reduces omen bad luck if nobody's nearby.

I LOVE this quirk, but trying to do antything at all except 'Suffer Miserably' is nigh impossible. To alleviate life a little, making it so that you have a lesser chance of suffering misfortune if nobody's around will be the perfect compromise. It makes life easier but doesn't compromise funny moments.

Any client in viewrange will disable the reduction. This includes ghosts.

## Changelog

:cl:
refactor: Adds charges to omens and omen smiting rather than only being permanent or one-use. Mirrors now grant seven bad luckers.
qol: Reduces omen bad luck if nobody's nearby to witness the funny. (Ghosts are included in the check!)
fix: Fixed an issue where a monkey check in doorcrushing was never actually able to pass. Also they screech now.
/:cl:

### CODE REVIEW

File: code/datums/components/omen.dm [hunk 1]
```diff
@@ -71,11 +93,23 @@
...
 		return
 
-	if(!prob(8 * luck_mod))
+	var/effective_luck = luck_mod
+
+	// If there's nobody to witness the misfortune, make it less likely.
+	// This way, we allow for people to be able to get into hilarious situations without making the game nigh unplayable most of the time.
+
+	var/has_watchers
+	for(var/mob/M in view(world.view, get_turf(src)))
```
Comment (san7890) [inline code/datums/components/omen.dm:105]: this is not hot enough to warrant a single letter variable, write it out

**Review by san7890 — CHANGES_REQUESTED** [review]

File: code/datums/components/omen.dm (hunk 1 above)
Comment (carlarctg) [reply code/datums/components/omen.dm:105]: wait, what? single letter vars help perf????

Comment (san7890) [reply code/datums/components/omen.dm:105]: no they don't but some people do it like that because they think it will.

Comment (san7890) [reply code/datums/components/omen.dm:105]: you should still fix the single letter var

File: code/datums/components/omen.dm [hunk 2]
```diff
@@ -71,11 +93,23 @@
...
+		if(!M.client)
+			continue
+		has_watchers = TRUE
+	if(has_watchers == FALSE)
+		effective_luck *= 0.5
```
Comment (Ghommie) [inline code/datums/components/omen.dm:110]:
```suggestion
	for(var/mob/viewer in view(world.view, get_turf(src)))
		if(viewer.client)
			effective_luck *= 0.5
			break
```

File: code/datums/components/omen.dm [hunk 3]
```diff
@@ -71,11 +93,20 @@
...
-	if(!prob(8 * luck_mod))
+	var/effective_luck = luck_mod
+
+	// If there's nobody to witness the misfortune, make it less likely.
+	// This way, we allow for people to be able to get into hilarious situations without making the game nigh unplayable most of the time.
+
+	for(var/mob/viewer in viewers(our_guy, world.view))
+		if(viewer.client)
+			effective_luck *= 0.5
+			break
```
Comment (Ghommie) [inline code/datums/components/omen.dm:107]:
```suggestion
	has_watchers = FALSE
	for(var/mob/viewer in viewers(our_guy, world.view))
		if(viewer.client)
			has_watchers = TRUE
			break
	if(!has_watchers)
		effective_luck *= 0.5
```
I've read it wrong, my bad.

**Pull request merged**